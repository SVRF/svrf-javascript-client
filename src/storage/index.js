import storageLocal, {LocalStorage} from './local';
import memoryStorage, {MemoryStorage} from './memory';

const isMemoryStorage = typeof localStorage === 'undefined';

/** @constant Storage class for creating custom storage */
export const Storage = isMemoryStorage ? MemoryStorage : LocalStorage;

/** @constant {Storage} App token info storage */
const appTokenStorage = isMemoryStorage ? memoryStorage : storageLocal;

export default appTokenStorage;
