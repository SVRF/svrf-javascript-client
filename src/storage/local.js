class LocalStorage {
  static LOCAL_STORAGE_KEY = 'svrf-app-token';

  static get() {
    const rawValue = localStorage.getItem(LocalStorage.LOCAL_STORAGE_KEY);
    return JSON.parse(rawValue);
  }

  static set(value) {
    const rawValue = JSON.stringify(value);
    localStorage.setItem(LocalStorage.LOCAL_STORAGE_KEY, rawValue);
  }

  static clear() {
    localStorage.removeItem(LocalStorage.LOCAL_STORAGE_KEY);
  }
}

export default LocalStorage;
