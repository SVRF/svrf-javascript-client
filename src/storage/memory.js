/**
 * App token info storage which uses memory
*/
export class MemoryStorage {
  /** @private */
  data = null;

  /**
   * Get app token info from variable
   * @returns {Object} App token info
   */
  get() {
    return this.data;
  }

  /**
   * Set app token info in variable
   * @param {Object} value - App token info
   */
  set(value) {
    this.data = value;
  }

  /**
   * Remove app token info from variable
   */
  clear() {
    this.data = null;
  }
}

const appTokenStorage = new MemoryStorage();
export default appTokenStorage;
