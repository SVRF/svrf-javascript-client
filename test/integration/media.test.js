import SVRF from '../../src';
import storage from '../../src/storage';
import TokenService from '../../src/services/token';

const {SVRF_TEST_API_KEY} = process.env;
const tokenService = new TokenService(storage);

describe('authentication', () => {
  const api = new SVRF(SVRF_TEST_API_KEY);

  beforeAll(async () => {
    await api.auth.authenticate();
  });

  afterEach(() => {
    tokenService.clearAppTokenInfo();
  });

  describe('getting media by id', () => {
    it('gets media if id is a string', async () => {
      const mediaId = '5431660791201792';
      const result = await api.media.getById(mediaId);

      expect(result.success).toEqual(true);
      expect(result.media.id).toEqual(mediaId);
    });

    it('gets media if id is a number', async () => {
      const mediaId = 5431660791201792;
      const result = await api.media.getById(mediaId);

      expect(result.success).toEqual(true);
      expect(result.media.id).toEqual(mediaId.toString());
    });
  });

  describe('getting trending media', () => {
    it('gets media', async () => {
      const result = await api.media.getTrending();

      expect(result.success).toEqual(true);
      expect(result.media).toBeInstanceOf(Array);
    });

    it('gets media with options', async () => {
      const options = {
        size: 15,
        type: [SVRF.enums.mediaType.PHOTO, SVRF.enums.mediaType.VIDEO],
      };

      const result = await api.media.getTrending(options);

      expect(result.success).toEqual(true);
      expect(result.media.length).toEqual(options.size);
      result.media.forEach((media) => {
        expect(options.type).toContain(media.type);
      });
    });

    it('gets media if type option is not array', async () => {
      const options = {
        size: 20,
        type: SVRF.enums.mediaType.PHOTO,
      };

      const result = await api.media.getTrending(options);

      expect(result.success).toEqual(true);
      expect(result.media.length).toEqual(options.size);
      result.media.forEach((media) => {
        expect(options.type).toEqual(media.type);
      });
    });
  });

  describe('searching media', () => {
    const query = 'INVASION';

    it('searches media by query', async () => {
      const result = await api.media.search(query);

      expect(result.success).toEqual(true);
      expect(result.media.length).toBeTruthy();
    });

    it('searches media by query with options', async () => {
      const options = {
        size: 10,
        type: SVRF.enums.mediaType.VIDEO,
      };

      const result = await api.media.search(query, options);

      expect(result.success).toEqual(true);
      expect(result.media.length).toEqual(options.size);
      result.media.forEach((media) => {
        expect(media.type).toEqual(options.type);
      });
    });
  });
});
