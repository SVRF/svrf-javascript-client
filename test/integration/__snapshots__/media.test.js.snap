// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`media api getting media by id handles 404 error 1`] = `"Request failed with status code 404"`;
